dali/daliadmin/daliadmin.cpp

```diff
@@ -590,6 +591,46 @@ static void dfspart(const char *lname,IUserDescriptor *userDesc, unsigned partnu
 
 //=============================================================================
 
+static void setdfspartattr(const char *lname, unsigned partNum, const char *attr, const char *value, IUserDescriptor *userDesc)
+{
+    verifyex(partNum>=0);
```

Comment:
Should this be partNum >= 1 ?
If I specify partNum of 0 then it crashes because 
IDistributedFilePart &part = file->queryPart(partNum);
returns NULL.

Reply:
It is 0 based, queryPart(0) is correct..

So perhaps the file you were testing with was odd in some way.
I should add a check that num. parts > 0.

Perhaps you were testing with a superfile? I am not validating that..